src/twisted/internet/_resolver.py

```diff
@@ -169,6 +170,10 @@ def resolveHostName(self, resolutionReceiver, hostName, portNumber=0,
 
         @return: see interface
         """
+        # If it's Python 2, encode this to IDNA.
+        if not _PY3 and isinstance(hostName, unicode):
+            hostName = hostName.encode('idna')
```

Comment:
I agree with @glyph's point about removing the conditional and dealing with IDNA regardless of Python version